src/tests/ngraph_helpers/snippets_ngraph_functions/include/two_element_wise_function.hpp

```diff
@@ -0,0 +1,156 @@
+        constructor_validate_and_infer_types();
+    }
+
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override {
+        return std::make_shared<DummyOperation1>(new_args.at(0), new_args.at(1), this->get_output_type());
+    }
+};
+
+class DummyOperation2 : public BaseDummyOperation {
+public:
+    OPENVINO_OP("DummyOperation2", "test::snippets");
+
+    DummyOperation2(
+        const Output<Node>& arg0,
+        const Output<Node>& arg1,
+        const element::Type& output_type = element::undefined) : BaseDummyOperation(arg0, arg1, output_type) {
+        constructor_validate_and_infer_types();
+    }
+
+    DummyOperation2(const BaseDummyOperation& op) : DummyOperation2(op.get_input_source_output(0), op.get_input_source_output(1), op.get_output_type()) {
+        constructor_validate_and_infer_types();
+    }
+
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override {
+        return std::make_shared<DummyOperation2>(new_args.at(0), new_args.at(1), this->get_output_type());
+    }
+};
+
+class TwoElementWiseFunctionParams {
+public:
+    class Actual {
+    public:
+        std::pair<element::Type, element::Type> convertion_before_op1;
+        element::Type convertion_before_op2_1;
+        std::pair<element::Type, element::Type> convertion_before_op2_2;
+        element::Type convertion_after_op2;
+    };
+
+    class Expected {
+    public:
+        std::pair<element::Type, element::Type> convertion_before_op1;
+        element::Type convertion_before_op2_1;
+        std::pair<element::Type, element::Type> convertion_before_op2_2;
+        element::Type convertion_after_op2;
+        element::Type convertion_before_result;
+    };
+};
+
+/**
+ * @class TwoElementWiseFunction
+ * @brief The function for unit tests only, includes dummy operations.
+ */
+class TwoElementWiseFunction : public SnippetsFunctionBase {
+public:
+    explicit TwoElementWiseFunction(
+        const std::vector<PartialShape> input_shapes,
+        const ngraph::element::Type precision1,
+        const ngraph::element::Type precision2,
+        const ngraph::element::Type constant_precision,
+        TwoElementWiseFunctionParams::Actual actual,
+        TwoElementWiseFunctionParams::Expected expected) :
+        SnippetsFunctionBase(input_shapes),
+        precision1(precision1),
+        precision2(precision2),
+        constant_precision(constant_precision),
+        actual(actual),
+        expected(expected) {
+        OPENVINO_ASSERT(input_shapes.size() == 2ull, "input_shapes size has to be equal to 2");
+    }
+
+    /*
+     * Don't call this method explicity. You should create the instance of PrecisionPropagationAddFunction before.
+     * After the method will be called implicitly in getOriginal or getReference methods.
+     * Note, please, getLowered method is not implemented and throws exception.
+     */
+    static std::shared_ptr<ngraph::Function> get(
```

Comment:
If the method shouldn't be called explicitly,then please make it private

Reply:
done